### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by lauzadis**
Upgrade third party dependencies to their latest verisons
## Issue \#

N/A
## Description of changes

The Kotlin SDK continues to aggressively take the latest versions of dependencies while we are in developer preview.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-runtime/aws-config/jvm/src/aws/sdk/kotlin/runtime/auth/credentials/executeCommandJVM.kt [hunk 1]
```diff
@@ -65,8 +65,11 @@ internal actual suspend fun executeCommand(
...
         Pair(
             process.exitValue(),
-            if (process.exitValue() == 0) output.toString()
-            else process.errorStream.bufferedReader().use { it.readText() },
+            if (process.exitValue() == 0) {
+                output.toString()
+            } else {
+                process.errorStream.bufferedReader().use { it.readText() }
+            },
         )
```
Comment (ianbotsf) [inline aws-runtime/aws-config/jvm/src/aws/sdk/kotlin/runtime/auth/credentials/executeCommandJVM.kt:73]: **Comment**: Multiline `if`-`else` block is a bit much for an inline argument. Consider a temp val outside the `Pair` call. Bonus: then you can eliminate the explicit `Pair` construction and just use [`to`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/to.html) instead:

```kotlin
val processOutput = when {
    process.exitValue() == 0 -> output.toString()
    else -> process.errorStream.bufferedReader().use { it.readText() }
}

process.exitValue() to processOutput // Pair(process.exitValue(), processOutput)
```

File: aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/config/profile/AWSConfigLoaderFilesystemTest.kt [hunk 2]
```diff
@@ -32,13 +32,13 @@ class AWSConfigLoaderFilesystemTest {
 
     @TempDir
     @JvmField
-    val tempDir: Path? = null
+    var tempDir: Path? = null
```
Comment (ianbotsf) [inline aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/config/profile/AWSConfigLoaderFilesystemTest.kt:35]: **Question**: Why is this change necessary?

**Review by ianbotsf — APPROVED** [review]

File: aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/config/profile/AWSConfigLoaderFilesystemTest.kt (hunk 2 above)
Comment (lauzadis) [reply aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/config/profile/AWSConfigLoaderFilesystemTest.kt:35]: In the upgraded version of JUnit, `@TempDir` fields cannot be `final`: https://github.com/junit-team/junit5/issues/2839

Comment (ianbotsf) [reply aws-runtime/aws-config/jvm/test/aws/sdk/kotlin/runtime/config/profile/AWSConfigLoaderFilesystemTest.kt:35]: That's too bad. In that case, just remove the `requireNotNull(tempDir)` checks in the unit tests. They're no longer doing anything meaningful now that the field is a `var`.
